### ISSUE TITLE
Too many arguments for method getting kwags & block

### ISSUE THREAD

**Issue by ojab** [castwide/solargraph#372]
```ruby
# @param error_message [String] string
def with_lock(error_message:, &block)
end

with_lock(error_message: 'error', &block)
```

```
$ solargraph typecheck
/tmp/tst/tst.rb:6 - Too many arguments to #with_lock
1 problem found.
```

**Comment by ojab** [comment]
Also we can omit block in params declaration:
```ruby
# @param error_message [String] string
def with_lock(error_message:)
  yield
end

with_lock(error_message: 'error', &block)
```

**Comment by castwide** [comment]
Confirmed. It'll be fixed in the next release.

**Comment by castwide** [comment]
Fixed in version 0.40.0.
